libcudacxx/test/libcudacxx/cuda/try_cancel/try_cancel.pass.cpp

```diff
@@ -0,0 +1,220 @@
+
+// UNSUPPORTED: pre-sm-100
+// UNSUPPORTED: !nvcc
+// UNSUPPORTED: nvrtc
+// UNSUPPORTED: c++98, c++03, c++14, c++17
+
+#include <cuda/try_cancel>
+
+__device__ void vec_add_impl1(int* a, int* b, int* c, int n, dim3 tb)
+{
+  int idx = threadIdx.x + tb.x * blockDim.x;
+  if (idx < n)
+  {
+    c[idx] += a[idx] + b[idx];
+  }
+}
+
+__device__ void vec_add_impl2(int* a, int* b, int* c, int n, dim3 tb)
+{
+  int x_idx = threadIdx.x + (tb.x * blockDim.x);
+  int y_idx = threadIdx.y + (tb.y * blockDim.y);
+  int idx   = x_idx + y_idx * (blockDim.x * gridDim.x);
+  if (idx < n)
+  {
+    c[idx] += a[idx] + b[idx];
+  }
+}
+
+__device__ void vec_add_impl3(int* a, int* b, int* c, int n, dim3 tb)
+{
+  int x_idx = threadIdx.x + (tb.x * blockDim.x);
+  int y_idx = threadIdx.y + (tb.y * blockDim.y);
+  int z_idx = threadIdx.z + (tb.z * blockDim.z);
+  int idx   = x_idx + y_idx * (blockDim.x * gridDim.x) + z_idx * (blockDim.x * gridDim.x * blockDim.y * gridDim.y);
+  if (idx < n)
+  {
+    c[idx] += a[idx] + b[idx];
+  }
+}
+
+__global__ void vec_add_det1(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::__detail::__for_each_cancelled_block<1>(threadIdx.x == tidx, [=](dim3 tb) {
+    vec_add_impl1(a, b, c, n, tb);
+  });
+}
+
+__global__ void vec_add_det2(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::__detail::__for_each_cancelled_block<2>(threadIdx.x == tidx && threadIdx.y == tidx, [=](dim3 tb) {
+    vec_add_impl2(a, b, c, n, tb);
+  });
+}
+
+__global__ void vec_add_det3(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::__detail::__for_each_cancelled_block<3>(
+    threadIdx.x == tidx && threadIdx.y == tidx && threadIdx.z == tidx, [=](dim3 tb) {
+      vec_add_impl3(a, b, c, n, tb);
+    });
+}
+
+__global__ void vec_add1(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::for_each_cancelled_block<1>([=](dim3 tb) {
+    vec_add_impl1(a, b, c, n, tb);
+  });
+}
+
+__global__ void vec_add2(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::for_each_cancelled_block<2>([=](dim3 tb) {
+    vec_add_impl2(a, b, c, n, tb);
+  });
+}
+
+__global__ void vec_add3(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::for_each_cancelled_block<3>([=](dim3 tb) {
+    vec_add_impl3(a, b, c, n, tb);
+  });
+}
+
+#include <iostream>
+#include <vector>
+
+template <typename F>
+bool test(size_t N, F&& f)
+{
+  for (int tidx : {0, 33, 63, 94})
+  {
+    int *a, *b, *c;
+    cudaMallocManaged(&a, N * sizeof(int));
+    cudaMallocManaged(&b, N * sizeof(int));
+    cudaMallocManaged(&c, N * sizeof(int));
+    for (int i = 0; i < N; ++i)
+    {
+      a[i] = i;
+      b[i] = 1;
+      c[i] = 0;
+    }
+
+    f(a, b, c, N, tidx);
+    if (auto e = cudaDeviceSynchronize(); e != cudaSuccess)
+    {
+      std::cerr << "ERROR: synchronize failed" << std::endl;
+      return false;
+    }
```

Comment:
I think in a unit test we can just assert:
```suggestion
    assert(cudaDeviceSynchronize() == cudaSuccess);
```
Maybe we already have a shorthand macro for this. @miscco ?

Reply:
I think @miscco already fixed this. Resolving (re-open if I got that wrong).